docs/DeveloperGuide.md

```diff
@@ -779,6 +781,17 @@ Preconditions: An undoable command has been executed by FABook.
 
 Use case ends.
 
+**Extensions**
+* 1a. No previous commands available.
+    * 1a1. FABook shows an error message that there are no commands to undo.
+
+  Use case ends.
+
+* 1b. The previous command is not undoable.
+    * 1b1. FABook shows an error message that there are no commands to undo.
+
+  Use case ends.
+
```

Comment:
Violates the precondition mentioned above, can just remove the preconditon ba

Reply:
I decided to remove precondition. Cause otherwise the precondition has to be "the command right before `undo` is undoable" and that might not be what we can reasonably expect.